### PR TITLE
Fixes for security issues regarding incomplete-string-escaping or encoding 

### PR DESCRIPTION
**Pull request by MottMac-PH**
Fixes for security issues regarding incomplete-string-escaping or encoding

### CODE REVIEW

File: SiteAssets/resources/pnp.js [hunk 1]
```diff
@@ -127,7 +127,7 @@ var Util = /** @class */ (function () {
      * @param name The name of the url paramter to check
      */
     Util.urlParamExists = function (name) {
-        name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+        name = name.replace(/\\/g, "\\\\").replace(/\[/g, "\\[").replace(/\]/g, "\\]");
```
Comment (JHolland98) [inline SiteAssets/resources/pnp.js:130]: Just want to make sure that we should be replacing back slashes with double backslashes, this part of the code: replace(/\\/g, "\\\\")

File: SiteAssets/resources/pnp.js [hunk 2]
```diff
@@ -137,7 +137,7 @@ var Util = /** @class */ (function () {
      * @param name The name of the paramter for which we want the value
      */
     Util.getUrlParamByName = function (name) {
-        name = name.replace(/[\[]/, "\\[").replace(/[\]]/, "\\]");
+        name = name.replace(/\\/g, "\\\\").replace(/\[/g, "\\[").replace(/\]/g, "\\]");
```
Comment (JHolland98) [inline SiteAssets/resources/pnp.js:140]: Same comment as above

**Review by JHolland98 — COMMENTED** [review]
Couple of comments in the pnp file

File: SiteAssets/resources/pnp.js (hunk 1 above)
Comment (MottMac-PH) [reply SiteAssets/resources/pnp.js:130]: The additional backslash is used to ensure that any backslashes in the original string are properly escaped, preventing them from being misinterpreted as escape characters.

**Comment by MottMac-PH** [comment]
Thanks for reviewing @JHolland98 , I have posted responses to your comments, hope they are acceptable. 
Let me know if anything else.

**Review by JHolland98 — APPROVED** [review]

**Pull request merged**